changelogs/fragments/8979-keycloak_group-fix-subgroups.yml

```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - keycloak_group - fix crash caused in subgroup creation. The crash was caused by a missing or empty `subGroups` property in Keycloak ≥23
```

Comment:
```suggestion
  - keycloak_group - fix crash caused in subgroup creation. The crash was caused by a missing or empty ``subGroups`` property in Keycloak ≥23 (https://github.com/ansible-collections/community.general/issues/8788, https://github.com/ansible-collections/community.general/pull/8979).
```

Reply:
Thank for your quick comment.

> Regarding the tests: unsupported means they don't run in CI, you have to run them manually.

Thanks for the explanation

> See also tests/integration/targets/keycloak_group/readme.adoc

I *may* have forgotten to read this. I have read it now.

> Did you try passing --python 3.12 explicitly?

That did the trick, kind of. I think my issue is that I'm on WSL. I also had to remove the dependency `setup_docker` to run the tests locally.

<details>
<summary>`ansible-test integration keycloak_group --allow-unsupported  --python 3.12 -vvv` [normal dependency] [truncated]</summary>

```
TASK [setup_docker : Install docker] *******************************************
task path: /home/gaudardv/ansible_collections/community/general/tests/output/.tmp/integration/keycloak_group-2rqbftlf-ÅÑŚÌβŁÈ/tests/integration/targets/setup_docker/tasks/default.yml:10
Using module file /tmp/ansible-test-ujtpqbcs/ansible/modules/apt.py
Pipelining is enabled.
<testhost> ESTABLISH LOCAL CONNECTION FOR USER: gaudardv
<testhost> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-nvhzecpkvqoanzhdevfiukxgizhqrlyg ; /usr/bin/python3'"'"' && sleep 0'
failed: [testhost] (item=docker.io) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "docker.io",
    "module_stderr": "sudo: a password is required\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```
</details>

Without this dependency, I was able to run the tests before and after my changes, and both passed.

Reply:
Hi @vgaudard 

You might want give it a try to `andebox` by yours truly:

https://pypi.org/project/andebox/

Tests that depend on `setup_docker` might be easier to execute using an accompanying VM. I have no idea if that would work in WSL, but locally what I would do is:

```
$ andebox vagrant -n ubuntu-jammy -s -- unsupported/keycloak_group --color yes -vvv
```